trl/trainer/rloo_trainer.py

```diff
@@ -1690,28 +1536,34 @@ def _generate_and_score_completions(
         # Apply weights to each reward function's output and sum
         rewards = (rewards_per_func * self.reward_weights.to(device).unsqueeze(0)).nansum(dim=1)
 
-        # Compute grouped-wise rewards
-        mean_grouped_rewards = rewards.view(-1, self.num_generations).mean(dim=1)
-
-        # Normalize the rewards to compute the advantages
-        mean_grouped_rewards = mean_grouped_rewards.repeat_interleave(self.num_generations, dim=0)
-        advantages = rewards - mean_grouped_rewards
-
-        if self.scale_rewards in ["batch", "none"]:
-            # If self.scale_rewards = "none", we'll still log group level std
-            std_rewards = rewards.view(-1, self.num_generations).std(dim=1)
-            std_rewards = std_rewards.repeat_interleave(self.num_generations, dim=0)
-        elif self.scale_rewards == "group":
-            # Compute global std
-            std_rewards = rewards.std().expand_as(rewards)
-        else:
-            raise ValueError(
-                f"Invalid value for scale_rewards: {self.scale_rewards}. Must be one of 'batch', 'group', or 'none'."
-            )
-
-        is_std_zero = torch.isclose(std_rewards, torch.zeros_like(std_rewards))
-        if self.scale_rewards in ["batch", "none"]:
-            advantages = advantages / (std_rewards + 1e-4)
+        # For RLOO, include sequence-level KL penalty in the reward itself (following original RLOO implementation)
+        if self.beta != 0.0 and ref_per_token_logps is not None and old_per_token_logps is not None:
```

Comment:
Is it possible to have `(self.beta != 0.0) != (ref_per_token_logps is not None)`?

Reply:
if you pass beta=0, ref log probs aren't computed